### PR TITLE
Implement a screen/action locker feature

### PR DESCRIPTION
**Pull request by nirvn**
This PR implements a screen locker in QField which, when enabled, block all (touch, mouse, stylus) actions. To enable it, select "Lock Screen" in the main menu (accessible by opening the dashboard or long-pressing the dashboard button).

image

Once enabled, QField will conveniently get rid of all button overlays on the canvas, which enables this feature to be extra helpful when QField is set to lock on the position (i.e., you'll see the map canvas, it'll pan around when the position is about to get out of the canvas extent, and you won't have any buttons overlaying your beautiful map.

image

To unlock, simply drag on a lock icon that appears at the center of the screen towards its unlock threshold, et voila:

https://user-images.githubusercontent.com/1728657/197777717-939c769b-f6a8-47bf-9bb1-c97adb1126c7.mp4

_Funded by [Groupements Forestiers Québec](https://groupementsforestiers.quebec/)_

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 9bcacff431b34114924bb1ac4a6235c40a1d275a: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3507-093a6f6f03eb692c299bacb3f9c42b5aaa1d97a2-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3507-093a6f6f03eb692c299bacb3f9c42b5aaa1d97a2-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3507-093a6f6f03eb692c299bacb3f9c42b5aaa1d97a2-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3507-093a6f6f03eb692c299bacb3f9c42b5aaa1d97a2-x86-android.apk)
</details>

File: src/qml/ScreenLocker.qml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  anchors.fill: parent
+  visible: enabled
+
+  onEnabledChanged: {
+    if (enabled) {
+        unlockNode.opacity = 1
+        unlockOpacityTimer.restart()
+    } else {
+        unlockNode.opacity = 0
+    }
```
Comment (suricactus) [inline src/qml/ScreenLocker.qml:21]:
```suggestion
    if (enabled) {
      unlockNode.opacity = 1
      unlockOpacityTimer.restart()
    } else {
      unlockNode.opacity = 0
    }
```

File: src/qml/ScreenLocker.qml [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+    Image {
+      anchors.centerIn: parent
+      width: 36
+      height: 36
+      fillMode: Image.PreserveAspectFit
+      smooth: true
+      source: unlockHandler.unlocked
+              ? Theme.getThemeVectorIcon('ic_lock_open_green_24dp')
+              : unlockHandler.entered ? Theme.getThemeVectorIcon('ic_lock_open_white_24dp') : Theme.getThemeVectorIcon('ic_lock_white_24dp')
```
Comment (suricactus) [inline src/qml/ScreenLocker.qml:105]:
```suggestion
      source: unlockHandler.unlocked
              ? Theme.getThemeVectorIcon('ic_lock_open_green_24dp')
              : unlockHandler.entered
                ? Theme.getThemeVectorIcon('ic_lock_open_white_24dp')
                : Theme.getThemeVectorIcon('ic_lock_white_24dp')
```

File: src/qml/ScreenLocker.qml [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        unlockNode.offsetX = 0
+        unlockNode.offsetY = 0
+      }
+    }
+
+    onCentroidChanged: {
+      if (entered) {
+        var x = centroid.position.x - centerX
+        var y = centroid.position.y - centerY
+        var distance = Math.sqrt(Math.pow(x,2) + Math.pow(y,2))
```
Comment (suricactus) [inline src/qml/ScreenLocker.qml:154]:
```suggestion
        var distance = Math.sqrt(Math.pow(x, 2) + Math.pow(y, 2))
```

File: src/qml/ScreenLocker.qml [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    property double centerX: parent.width / 2
+    property double centerY: parent.height / 2
+    property bool entered: false
+    property bool unlocked: false
+
+    onActiveChanged: {
+      if (active) {
+        var x = centroid.position.x - centerX
+        var y = centroid.position.y - centerY
+        var distance = Math.sqrt(Math.pow(x,2) + Math.pow(y,2))
```
Comment (suricactus) [inline src/qml/ScreenLocker.qml:132]:
```suggestion
        var distance = Math.sqrt(Math.pow(x, 2) + Math.pow(y, 2))
```

**Review by suricactus — APPROVED** [review]
Excellent code, just turned on my linter mode. Will try it later.

**Pull request merged**